analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/CalculationsShouldNotOverflowBase.cs

```diff
@@ -0,0 +1,29 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
+
+public abstract class CalculationsShouldNotOverflowBase : SymbolicRuleCheck
+{
+    protected const string DiagnosticId = "S3949";
+    protected const string MessageFormat = "This calculation is {0} to overflow the maximum value of '{1}'.";
+    protected const string MessageFormatLikely = "likely";
+    protected const string MessageFormatGuaranteed = "guaranteed";
```

Comment:
These two are not `Format`-strings
```suggestion
    protected const string MessageLikely = "likely";
    protected const string MessageGuaranteed = "guaranteed";
```
or just this for simplicity
```suggestion
    protected const string Likely = "likely";
    protected const string Guaranteed = "guaranteed";
```